modules/ROOT/pages/policy-mule4-detokenization.adoc

```diff
@@ -0,0 +1,155 @@
+= Detokenization Policy
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+[%autowidth.spread,cols="a,a"]
+|===
+>s| Policy Name | Detokenization.
+>s| Summary      | Transforms a tokenized value back into the original data.
+>s| Category | Security.
+>s| Since Mule Version | 4.1.1
+.4+>.^s| Returned Status Codes |
+
+[%autowidth.spread,cols="a,a"]
+!===
+.2+>.^! 400 ! A token is invalid, or their detokenization failed.
+! The provided expression is invalid or returns no result to detokenize. +
+*NOTE*: The status code to be returned can be configured.
+! 502 ! Any problem regarding the connection with the tokenization service. It may be unavailable, or it might not have been configured.
+!===
+
+|===
+
+[NOTE]
+--
+This policy is designed to work only for Anypoint Runtime Fabric deployments.
+--
+
+== Tokenization
+
+Detokenization is the process of detokenizing a token back into its original value to redeem sensitive data to carry on with a specific business purpose.
+For example, an application would need to use a detokenized value to validate a specific credit card number. +
+Detokenized sensitive data must always be read under strict security controls.
+
+== Prerequisites
+
+* To create and apply policies you need the *Manage Policies* permission.
+* You need to have a tokenization service up and running.
+** To configure and see the tokenization service, you need the *Manage Tokenization Services* permission. +
+See xref:anypoint-security::tokenization-example.adoc[Configure and Use Tokenization] for more information.
+
+
+== How Does This Policy Work?
+
+The policy reads a token the payload with the tokenized values. To define which pieces of payload will be detokenized, you need to provide Dataweave 2.0 expressions.
```

Comment:
Please replace Dataweave 2.0 expressions with "selector expressions". We made some last minute modifications and we are not using Dataweave anymore.

Reply:
Understood. See: df71b31